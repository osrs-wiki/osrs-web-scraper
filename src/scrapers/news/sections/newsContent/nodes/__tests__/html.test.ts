import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import htmlParser from "../html";

describe("html node", () => {
  test("Tag should parse and render MediaWikiHTML", () => {
    const root = parse("<sub>test</sub>");
    const builder = new MediaWikiBuilder();
    builder.addContents([htmlParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
