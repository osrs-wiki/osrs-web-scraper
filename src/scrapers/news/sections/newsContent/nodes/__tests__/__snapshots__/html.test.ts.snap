// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`html node Tag should parse and render MediaWikiHTML 1`] = `"<sub>test</sub>"`;
